packages/integration-tests/projects/suite-web/tests/coinmarket/exchange.test.ts

```diff
@@ -0,0 +1,187 @@
+        cy.discoveryShouldFinish();
+        // Tests all input windows are empty
+        cy.getTestElement('@coinmarket/exchange/crypto-input').should('have.value', '');
+        cy.getTestElement('@coinmarket/exchange/fiat-input').should('have.value', '');
+
+        // Tests crypto input contains REGTEST
+        cy.getTestElement('@coinmarket/exchange/crypto-currency-select/input').should(
+            'contain.text',
+            'REGTEST',
+        );
+
+        // Fills out 0.005REGTEST and chooses ETH as target crypto */
+        cy.getTestElement('@coinmarket/exchange/crypto-input').type('0.005');
+        cy.getTestElement('@coinmarket/exchange/receive-crypto-select/input').type('ETH{enter}');
+
+        // Custom fee setup
+        cy.getTestElement('select-bar/custom').click();
+        cy.getTestElement('feePerUnit').clear().type('1');
+        cy.getTestElement('@coinmarket/exchange/compare-button').click();
+
+        // Verifies the offers displayed match the mock
+        cy.fixture('./invity/exchange/quotes').then((quotes: any) => {
+            const exchangeProvider = [
+                ['changehero', 'changehero'],
+                ['changenow', 'changenow'],
+                ['changelly', 'changelly'],
+            ];
+
+            exchangeProvider.forEach((provider: string[]) => {
+                // Tests offer accordance with the mocks
+                const valueFromFixtures = quotes.find(
+                    (quote: any) => quote.exchange === provider[1],
+                );
+                cy.contains('[class*="Quote__Wrapper"]', provider[0], { matchCase: false })
+                    .should('exist')
+                    .find('[class*="CryptoAmount__Value"]') // Returns element handle
+                    .invoke('text')
+                    .then((readValue: string) => {
+                        const ethValueFromApp: number = parseFloat(readValue);
+                        const ethValueFromQuote: number = parseFloat(
+                            valueFromFixtures.receiveStringAmount,
+                        );
+                        expect(ethValueFromApp).to.be.eq(ethValueFromQuote);
+                    });
+            });
+        });
+
+        // Gets the deal
+        cy.getTestElement('@coinmarket/exchange/offers/get-this-deal-button').eq(0).click();
+        cy.getTestElement('@modal').should('be.visible');
+        cy.getTestElement('@coinmarket/exchange/offers/buy-terms-agree-checkbox').click();
+        cy.getTestElement('@coinmarket/exchange/offers/buy-terms-confirm-button').click();
+
+        // Verifies amounts, currencies and providers
+        cy.get('[class*="CoinmarketExchangeOfferInfo__Wrapper"]')
+            .should('exist')
+            .then(wrapper => {
+                cy.wrap(wrapper)
+                    .find('[class*="FormattedCryptoAmount__Value"]')
+                    .first()
+                    .invoke('text')
+                    .should('be.equal', '0.005');
```

Comment:
Move all inputed values into an object at the beginning of the test and use references withing the test itself. It improves readability and helps with reusage. For reference, check eg `bug-report-form.test.ts`